### ISSUE TITLE
`accessibility_trait_for_button` false positive with `if #available` check

### ISSUE THREAD

**Issue by hbiede** [realm/SwiftLint#6446]
### New Issue Checklist

- [X] I've Updated SwiftLint to the latest version.
- [X] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

Using `. isToggle` in an app that supports a version before iOS 17 requires an `if #available`, which breaks this lint rule. Example:

```swift
struct ContentView: View {
    var body: some View {
        Text("Hello, World!")
            .onTapGesture {
                print("foo")
            }
            .accessibilityAddTraits({
                if #available(iOS 17, *) {
                    [.isToggle, .isButton]
                } else {
                    .isButton
                }
            }())
    }
}
```

Mention the command or other SwiftLint integration method that caused the issue. Include stack traces or command output.

```bash
$ swiftlint lint --strict
Linting Swift files in current working directory
Linting 'TestApp.swift' (2/2)
Linting 'ContentView.swift' (1/2)
~/Desktop/test/test/ContentView.swift:12:9: error: Accessibility Trait for Button Violation: All views with tap gestures added should include the .isButton or the .isLink accessibility traits (accessibility_trait_for_button)
Done linting! Found 1 violation, 1 serious in 2 files.
```

### Environment

* SwiftLint version:
```bash
$swiftlint version
0.63.1
```
* Xcode version:
```bash
$xcodebuild -version
Xcode 26.2
Build version 17C52
```
* Installation method used: Homebrew
* Configuration file:

```yml
# Ran in minimal setup to test
opt_in_rules:
  - accessibility_trait_for_button
```

Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? No

**Comment by hbiede** [comment]
In reviewing the history for the rule file, it seems to have come after the [SourceKit migration](https://github.com/realm/SwiftLint/commit/5eac9be50dc2196bc29a425e6fa46518ffbedc80)

In looking at the code, my guess is that [this condition](https://github.com/realm/SwiftLint/blob/afc957ab8cbf509849f54cfa0f743d4d4aed3536/Source/SwiftLintBuiltInRules/Rules/Lint/AccessibilityTraitForButtonRule.swift#L257) should have an else block to check if all results of a condition contain the appropriate traits (as this would also handle things like `isHeader ? [.isButton, .isHeader] : [.isButton]`)

**Comment by SimplyDanny** [comment]
The rule currently only checks for the most obvious cases and even if we extended the logic, it would never be complete. Easiest example is `.accessibilityAddTraits(traits)`. `traits` can be everything, we just don't know.

We can keep the issue open for anyone who would like to add more coverage. But this would only ever be a case-by-case extension.
